rero_mef/agents/api.py

```diff
@@ -0,0 +1,249 @@
+                                        dbcommit=False, reindex=False,
+                                        test_md5=False, online=True,
+                                        verbose=False):
+        """Create or update agent, Mef and Viaf record."""
+        try:
+            persistent_id = PersistentIdentifier.query.filter_by(
+                pid_type=cls.provider.pid_type,
+                pid_value=data.get('pid')
+            ).one()
+            if persistent_id.status == PIDStatus.DELETED:
+                return None, Action.ALREADYDELETED, None, Action.DISCARD, \
+                    None, False
+        except Exception as err:
+            pass
+
+        record, action = cls.create_or_update(
+            data=data,
+            id_=id_,
+            delete_pid=delete_pid,
+            dbcommit=dbcommit,
+            reindex=reindex,
+            test_md5=test_md5
+        )
+        if record.deleted:
+            mef_record, mef_action = record.delete_from_mef(
+                dbcommit=dbcommit,
+                reindex=reindex,
+                verbose=verbose
+            )
+            record.delete(
+                dbcommit=dbcommit,
+                delindex=True,
+            )
+            record = None
+            action = Action.DELETE
+            viaf_record = None
+            online = False
+        else:
+            mef_record, mef_action, viaf_record, online = \
+                record.create_or_update_mef_viaf_record(
+                    dbcommit=dbcommit,
+                    reindex=reindex,
+                    online=online
+                )
+        return record, action, mef_record, mef_action, viaf_record, online
+
+    def delete_from_mef(self, dbcommit=False, reindex=False, verbose=False):
+        """Delete agent from MEF record."""
+        from ..mef.api import MefRecord
+        action = Action.DISCARD
+        mef_pid = 'Non'
+        mef_record = MefRecord.get_mef_by_agent_pid(self.pid, self.agent)
+        if mef_record:
+            mef_pid = mef_record.pid
+            if mef_record.pop(self.agent, None):
+                action = Action.DELETEAGENT
+                mef_record = mef_record.replace(
+                    data=mef_record,
+                    dbcommit=dbcommit,
+                    reindex=reindex
+                )
+                if reindex:
+                    MefRecord.update_indexes()
+        if verbose:
+            click.echo(
+                'Delete {agent}: {pid} from mef: {mef_pid} {action}'.format(
+                    agent=self.agent,
+                    pid=self.pid,
+                    mef_pid=mef_pid,
+                    action=action.value
+                )
+            )
+        return mef_record, action
+
+    @classmethod
+    def get_online_record(cls, id, verbose=False):
+        """Get online Record.
+
+        Has to be overloaded in agent class.
+        """
+        return None
```

Comment:
By default, a python function return `None` if no return statement is defined. In this case, you could just use `pass` statement to replace `return None`.

Regarding the function docstring, maybe it will be a good thing if this method `raise NotYetImplementedMethod()` ? (This class will be a kind of java abstract class)